File: app/src/main/java/com/soft/credit911/ForgotPassword/ForgetPasswordActivity.java
package com.soft.credit911.ForgotPassword;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.soft.credit911.ForgotPassword.model.ForgotPasswordResponse;
import com.soft.credit911.ForgotPassword.mvp.ForgotPasswordPresenter;
import com.soft.credit911.ForgotPassword.mvp.ForgotPasswordView;
import com.soft.credit911.NetworkUtils.APIConstants;
import com.soft.credit911.R;
import com.soft.credit911.Utils.AppConstants;
import com.soft.credit911.Utils.CommonUtils;
import com.soft.credit911.databinding.ActivityForgetPasswordBinding;
import com.soft.credit911.databinding.ActivityMainBinding;
import com.soft.credit911.databinding.ToolbarBinding;

import java.security.PrivateKey;

import static com.soft.credit911.Utils.CommonUtils.isValidEmail;

public class ForgetPasswordActivity extends AppCompatActivity implements ForgotPasswordView {

    private ActivityForgetPasswordBinding layoutBinding;
    private ToolbarBinding toolbarBinding;
    private ForgotPasswordPresenter forgotPasswordPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        layoutBinding = ActivityForgetPasswordBinding.inflate(getLayoutInflater());
        toolbarBinding = layoutBinding.toolbarLayout;
        toolbarBinding.toolbarTitle.setText("Forgot Password");
        toolbarBinding.navigationIcon.setOnClickListener(v -> {
            onBackPressed();
        });
        View view = layoutBinding.getRoot();
        setContentView(view);
        forgotPasswordPresenter = new ForgotPasswordPresenter(this, this);
        layoutBinding.tvResetPassword.setOnClickListener(v -> {
            if (validate())
                forgotPasswordPresenter.forgotPassword(layoutBinding.etUserEmail.getText().toString().trim());
        });
    }

    private boolean validate() {
        if (layoutBinding.etUserEmail.getText().toString().trim().equals("")) {
            layoutBinding.etUserEmail.setError("Enter Your Email Id");
            layoutBinding.etUserEmail.requestFocus();
            return false;
        }

        if (!isValidEmail(layoutBinding.etUserEmail.getText().toString().trim())) {
            layoutBinding.etUserEmail.setError("Invalid Email Id");
            layoutBinding.etUserEmail.requestFocus();
            return false;
        }

        return true;
    }

    @Override
    public void forgotPasswordResponse(ForgotPasswordResponse forgotPasswordResponse) {
        CommonUtils.showdialog(forgotPasswordResponse.getMessage(), this, forgotPasswordResponse.getStatus().equals(AppConstants.API_SUCCESS));
//Toast.makeText(this, forgotPasswordResponse.getMessage(), Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/soft/credit911/ui/ForgotPassword/ForgetPasswordActivity.kt
package com.soft.credit911.ui.ForgotPassword

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.soft.credit911.Utils.AppConstants
import com.soft.credit911.Utils.CommonUtils
import com.soft.credit911.databinding.ActivityForgetPasswordBinding
import com.soft.credit911.databinding.ToolbarBinding
import com.soft.credit911.datamodel.ForgotPasswordResponse
import com.soft.credit911.ui.ForgotPassword.mvp.ForgotPasswordPresenter
import com.soft.credit911.ui.ForgotPassword.mvp.ForgotPasswordView

class ForgetPasswordActivity : AppCompatActivity(), ForgotPasswordView {
    private var layoutBinding: ActivityForgetPasswordBinding? = null
    private var toolbarBinding: ToolbarBinding? = null
    private var forgotPasswordPresenter: ForgotPasswordPresenter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        layoutBinding = ActivityForgetPasswordBinding.inflate(
            layoutInflater
        )
        toolbarBinding = layoutBinding!!.toolbarLayout
        toolbarBinding!!.toolbarTitle.text = "Forgot Password"
        toolbarBinding!!.navigationIcon.setOnClickListener { v: View? -> onBackPressed() }
        val view = layoutBinding!!.root
        setContentView(view)
        forgotPasswordPresenter = ForgotPasswordPresenter(this, this)
        layoutBinding!!.tvResetPassword.setOnClickListener { v: View? ->
            if (validate()) forgotPasswordPresenter!!.forgotPassword(
                layoutBinding!!.etUserEmail.text.toString().trim { it <= ' ' })
        }
    }

    private fun validate(): Boolean {
        if (layoutBinding!!.etUserEmail.text.toString().trim { it <= ' ' } == "") {
            layoutBinding!!.etUserEmail.error = "Enter Your Email Id"
            layoutBinding!!.etUserEmail.requestFocus()
            return false
        }
        if (!CommonUtils.isValidEmail(
                layoutBinding!!.etUserEmail.text.toString().trim { it <= ' ' })
        ) {
            layoutBinding!!.etUserEmail.error = "Invalid Email Id"
            layoutBinding!!.etUserEmail.requestFocus()
            return false
        }
        return true
    }

    override fun forgotPasswordResponse(forgotPasswordResponse: ForgotPasswordResponse) {
        CommonUtils.showdialog(
            forgotPasswordResponse.message,
            this,
            forgotPasswordResponse.status == AppConstants.API_SUCCESS
        )
        //Toast.makeText(this, forgotPasswordResponse.getMessage(), Toast.LENGTH_SHORT).show();
    }
}